src/lib/y2storage/devicegraph.rb

```diff
@@ -159,6 +165,15 @@ def safe_copy(devicegraph)
       true
     end
 
+    # Copies the devigraph into another one
+    #
+    # @param devicegraph [Devicegraph]
+    def copy(devicegraph)
```

Comment:
question: so, a devicegraph can be copied on itself with this method, right?

Reply:
Yes. The method `#copy` from the library does not prevent it. We added a `#safe_copy` to avoid that.